lib/new_relic/agent/instrumentation/mongo.rb

```diff
@@ -16,7 +16,44 @@
 
   executes do
     NewRelic::Agent.logger.info 'Installing Mongo instrumentation'
-    install_mongo_instrumentation
+    if NewRelic::Agent::Datastores::Mongo.is_monitoring_enabled?
+      install_mongo_command_subscriber
+    else
+      install_mongo_instrumentation
+    end
+  end
+
+  class CommandSubscriber
+
+    def started(event)
+      operations[event.operation_id] = event
+    end
+
+    def succeeded(event)
+      started_event = operations.delete(event.operation_id)
+      NewRelic::Agent.instance.transaction_sampler.notice_nosql_statement(
+        format_statement(started_event.command), event.duration
+      )
+    end
+
+    def failed(event)
+      started_event = operations.delete(event.operation_id)
+      # How does new relic want to handle failures?
+    end
+
+    private
+
+    def operations
+      @operations ||= {}
+    end
+
+    def format_statement(event)
+      "#{event.database_name}.#{event.command_name} #{event.command.inspect}"
```

Comment:
The existing instrumentation for 1.x has support for [controlling capture and obfuscation of the queries](https://github.com/newrelic/rpm/blob/master/lib/new_relic/agent/datastores/mongo/obfuscator.rb) based on a [config setting](https://github.com/newrelic/rpm/blob/master/lib/new_relic/agent/configuration/default_source.rb#L837-L850). Will probably want to behave according to those here as well.